### PR TITLE
Don't require masking for Contains checks to be affected by CornerRadius.

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/Containers/Container.cs [hunk 1]
```diff
@@ -621,7 +621,7 @@ protected override bool InternalContains(Vector2 screenSpacePos)
             float cRadius = CornerRadius;
 
             // Select a cheaper contains method when we don't need rounded edges.
-            if (!Masking || cRadius == 0.0f)
```
Comment (Tom94) [inline osu.Framework/Graphics/Containers/Container.cs]: We should have a testcase for this. Not having one caused you to forget also removing the Masking check in `BoundingBox` (or whatever it's called by now).

Comment (Tom94) [inline osu.Framework/Graphics/Containers/Container.cs]: Can probably duplicate one of the nested rounded boxes in the masking test with disabled masking.

**Comment by Tom94** [comment]
No test case? :(

**Comment by smoogipooo** [comment]
As discussed in #ops, test case will be written when the whole testing framework is reworked/rewritten.

**Pull request merged**